p-token/tests/setup/mod.rs

```diff
@@ -6,3 +6,46 @@ pub mod account;
 pub mod mint;
 
 pub const TOKEN_PROGRAM_ID: Pubkey = Pubkey::new_from_array(spl_token_interface::program::ID);
+
+#[macro_export]
+macro_rules! assert_instruction_error {
+    ( $error:expr, $matcher:pat ) => {
+        match $error {
+            solana_program_test::BanksClientError::TransactionError(
+                solana_sdk::transaction::TransactionError::InstructionError(_, $matcher),
+            ) => {
+                assert!(true)
+            }
+            err => assert!(false, "Expected instruction error but got '{:#?}'", err),
+        };
+    };
+}
```

Comment:
nit: Rather than rewriting a macro, users could probably use `assert_matches` instead for more clarity and control https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html